### PR TITLE
deprecate baremodule in favor of 'bare module' 

### PR DESCRIPTION
**Pull request by bicycle1885**
This deprecates `baremodule` in favor of a two-word phrase `bare module` in the similar way as `mutable struct`, `primitive type`, and `abstract type`. This is mainly for consistency. Also, this frees the `baremodule` keyword, which would be scarcely used.

This is built on top of #26491.

I haven't updated the code and docs yet. If you like the idea, I'm going to finish that soon.

### CODE REVIEW

**Comment by bicycle1885** [comment]
I think this is ready. The CI failures look false alarm.

File: base/show.jl [hunk 1]
```diff
@@ -926,7 +926,7 @@ function show_block(io::IO, head, args::Vector, body, indent::Int)
         end
     end
 
-    ind = head === :module || head === :baremodule ? indent : indent + indent_width
+    ind = head === :module || head === Symbol("bare module") ? indent : indent + indent_width
```
Comment (JeffBezanson) [inline base/show.jl]: I don't think this expression head ever occurs (true of `:baremodule` as well, I believe).

Comment (bicycle1885) [reply base/show.jl]: I think it is generated from this line: https://github.com/JuliaLang/julia/pull/26494/files/bb6996198aa841e60b5238b2269d979bb2c0d037#diff-b4267643a238b5a375b3e055817ee6caR1322.

**Comment by StefanKarpinski** [comment]
That's a good idea. "bare" was kind of an arbitrary word choice anyway.

**Comment by JeffBezanson** [comment]
It is a good idea, but it's a bit of a pun. A `primitive type` is a specific kind of data type, not a "simpler" type or a type with fewer features.

**Comment by bicycle1885** [comment]
Yeah, but bare modules are still kind of a "primitive" stuff, aren't they?. A dictionary says "primitive" means "not developed or derived from anything else", which I think is as fitting a name as "bare" is. I think nobody think `primitive module` and `primitive type` share some kind of common properties in Julia.

**Comment by JeffBezanson** [comment]
Triage thinks this is not really needed. This feature is used fairly rarely, and the `baremodule` keyword doesn't bother me so much. We already have multiple kinds of types, so the `X type` naming scheme makes sense, but if we had multiple kinds of modules in the future it's not clear that we'd want to use the same scheme.

**Pull request closed without merging**